controls/SV-257928.rb

```diff
@@ -17,15 +15,25 @@
 $ sudo chown root [Public Directory]'
   impact 0.5
   ref 'DPMS Target Red Hat Enterprise Linux 9'
-  tag check_id: 'C-61669r925769_chk'
   tag severity: 'medium'
+  tag gtitle: 'SRG-OS-000480-GPOS-00227'
   tag gid: 'V-257928'
   tag rid: 'SV-257928r925771_rule'
   tag stig_id: 'RHEL-09-232240'
-  tag gtitle: 'SRG-OS-000480-GPOS-00227'
   tag fix_id: 'F-61593r925770_fix'
-  tag satisfies: ['SRG-OS-000480-GPOS-00227', 'SRG-OS-000138-GPOS-00069']
-  tag 'documentable'
   tag cci: ['CCI-000366', 'CCI-001090']
   tag nist: ['CM-6 b', 'SC-4']
+  tag 'host'
+  tag 'container'
+
+  partitions = etc_fstab.params.map { |partition| partition['mount_point'] }.uniq
+
+  cmd = "find #{partitions.join(' ')} -xdev -type d -perm -0002 -uid +999 -print"
```

Comment:
Going back to my earlier comments asking if we actually need to call `-print` here:

> The -print action is performed on all files for which the whole expression is true, unless it contains an action other than -prune or -quit.  Actions which inhibit the default -print are -delete, -exec, -execdir, -ok, -okdir, -fls, -fprint, -fprintf, -ls, -print and -printf.

None of those are true here so we could probably get rid of the call to that parameter.  However, this is also in the man page: 

> If you are piping the output of find into another program and there is the faintest possibility that the files which you are searching for might contain a newline, then you should seriously consider using the -print0 option instead of -print.

Since this is explicitly what we're doing, should we swap all usages of print with print0?  The other code change we'd need to do is swap the character we're doing the split on from \n to \0 (the null character).

An argument against it is that in this specific instance anything that shows up is a failure, but what about the several controls when we're putting in exemptions from an input?  Is it possible that a filename could be constructed that could take advantage of this?

Doing some research, it seems like you can relatively easily append newlines to the end of a filename.  Splitting the results of -print on a newline would then allow for a malicious filename to get through.  

However, my non-exhaustive skim of the various finds we are doing didn't show anywhere that that could be taken advantage of unless the user's inputs file has been compromised in some manner or they've done a typo of some kind at which point we're boned regardless.

tl;dr you're probably better off not reading unless you like the musings of 3am brain.  it was at least fun to read some of the find manual.